dask_image/ndinterp/__init__.py

```diff
@@ -105,21 +107,21 @@ def affine_transform(
         offset = matrix[:image.ndim, image.ndim]
         matrix = matrix[:image.ndim, :image.ndim]
 
+    cval = kwargs.pop('cval', 0)
+    mode = kwargs.pop('mode', 'constant')
+    prefilter = kwargs.pop('prefilter', False)
+    if mode in ['wrap', 'reflect', 'mirror', 'grid-mirror', 'grid-wrap']:
+        raise NotImplementedError(f"Mode {mode} is not currently supported.")
+
     # process kwargs
-    # prefilter is not yet supported
-    if 'prefilter' in kwargs:
-        if kwargs['prefilter'] and order > 1:
-            warnings.warn('Currently, `dask_image.ndinterp.affine_transform` '
-                          'doesn\'t support `prefilter=True`. Proceeding with'
-                          ' `prefilter=False`, which if order > 1 can lead '
-                          'to the output containing more blur than with '
-                          'prefiltering.', UserWarning)
-        del kwargs['prefilter']
-
-    if 'mode' in kwargs:
-        if kwargs['mode'] in ['wrap', 'reflect', 'mirror']:
-            raise(NotImplementedError("Mode %s is not currently supported."
-                                      % kwargs['mode']))
+    if prefilter and order > 1:
+        # prefilter is not yet supported for all modes
+        if mode in ['nearest', 'grid-constant']:
+            raise NotImplementedError(
+                f"order > 1 with mode='{mode}' is not supported."
```

Comment:
Same here, it would be good to point users in the right direction (i.e. what modes *are* supported that they should try instead). It's nice to have error messages that say both what went wrong and how you might fix it.

Reply:
Yeah, given the other wrap, etc. modes that `affine_transform` itself doesn't work with, only 'constant' mode can currently perform the prefiltering.